packages/react/src/authentication/adapters/CredentialsStorageSchema.ts

```diff
@@ -0,0 +1,51 @@
+import { BaseStorageSchema, IStorageItem } from '@lens-protocol/storage';
+import jwtDecode from 'jwt-decode';
+import isObject from 'lodash/isObject';
+import { TypeOf, z } from 'zod';
+
+const AuthData = z.object({
+  refreshToken: z.string(),
+});
+
+export type AuthData = z.infer<typeof AuthData>;
+
+type V1JwtPayload = {
+  id: string;
+  role: 'refresh';
+  iat: number;
+  exp: number;
+};
+
+function isV1Content(decodedJwt: unknown): decodedJwt is V1JwtPayload {
+  return isObject(decodedJwt) && 'role' in decodedJwt && decodedJwt.role === 'refresh';
+}
+
+export class CredentialsStorageSchema extends BaseStorageSchema<typeof AuthData> {
```

Comment:
`AuthData` is already available as a type in scope at line 10.

```suggestion
export class CredentialsStorageSchema extends BaseStorageSchema<AuthData> {
```